go-packages/meep-model/nodeMap_test.go

```diff
@@ -0,0 +1,408 @@
+		context := make(map[string]string)
+		context["pl"] = "phyLoc"
+		context["nl"] = "netLoc"
+		nm.AddNode(NewNode(domain.Name, domain.Type_, domain, &domain.Zones, scenario.Deployment, context))
+		////fmt.Printf("  domain%d: object @ %p\n%+v\n", i, &scenario.Deployment.Domains[i], domain)
+		////fmt.Printf("  domain%d: child @ %p\n%+v\n", i, &domain.Zones, domain.Zones)
+		////fmt.Printf("  domain%d: parent @ %p\n%+v\n", i, scenario.Deployment, *scenario.Deployment)
+	}
+	n := nm.FindByName("PUBLIC")
+	////fmt.Printf("%+v\n", n)
+	if n.object != &scenario.Deployment.Domains[0] {
+		t.Errorf("Domain[0] wrong object reference ")
+	}
+	if n.child != &scenario.Deployment.Domains[0].Zones {
+		t.Errorf("Domain[0] wrong child reference ")
+	}
+	if n.parent != scenario.Deployment {
+		t.Errorf("Domain[0] wrong parent reference ")
+	}
+	n = nm.FindByName("operator1")
+	////fmt.Printf("%+v\n", n)
+	if n.object != &scenario.Deployment.Domains[1] {
+		t.Errorf("Domain[1] wrong object reference ")
+	}
+	if n.child != &scenario.Deployment.Domains[1].Zones {
+		t.Errorf("Domain[1] wrong child reference ")
+	}
+	if n.parent != scenario.Deployment {
+		t.Errorf("Domain[1] wrong parent reference ")
+	}
+	// Change an object field via the node
+	testID := "new-test-id"
+	objPtr := n.object.(*ceModel.Domain)
+	////fmt.Printf("  node.object ptr %p\n%+v\n", objPtr, *objPtr)
+	objPtr.Id = testID
+	if scenario.Deployment.Domains[1].Id != testID {
+		t.Errorf("Failed changing domain id")
+	}
+	// Change a child field via the node
+	childPtr := n.child.(*[]ceModel.Zone)
+	//fmt.Printf("  node.child ptr %p\n%+v\n", childPtr, *childPtr)
+	(*childPtr)[0].Id = testID
+	if scenario.Deployment.Domains[1].Zones[0].Id != testID {
+		t.Errorf("Failed changing zone[0] id")
+	}
+	// Change a parent field via the node
+	parentPtr := n.parent.(*ceModel.Deployment)
+	//fmt.Printf("  node.parent ptr %p\n%+v\n", parentPtr, *parentPtr)
+	parentPtr.InterDomainLatency = 500
+	if scenario.Deployment.InterDomainLatency != 500 {
+		t.Errorf("Failed changing Deployment InterDomainLatency")
+	}
+	// Verify Node context
+	context := n.context.(map[string]string)
+	if context["pl"] != "phyLoc" || context["nl"] != "netLoc" {
+		t.Errorf("Failed to set context entries")
+	}
+}
+
+func TestNodeMapZone(t *testing.T) {
+	fmt.Println("--- ", t.Name())
+	log.MeepTextLogInit(t.Name())
+
+	fmt.Println("Create a scenario structure")
+	scenario := new(ceModel.Scenario)
+	err := json.Unmarshal([]byte(testScenario), scenario)
+	if err != nil {
+		t.Errorf("Unable to unmarshall scenario")
+	}
+
+	fmt.Println("Create nodeMap")
+	nm := NewNodeMap()
+	if nm == nil {
+		t.Errorf("Unable to allocate nodeMap")
+	}
+
+	fmt.Println("Test Deployment")
+	if scenario.Deployment == nil {
+		t.Errorf("nil deployment")
+	}
+
+	fmt.Println("Test Zones")
+	domain := &scenario.Deployment.Domains[1]
+	//fmt.Printf("  scenario.Deployment.Domains[1] has %d zones\n", len(scenario.Deployment.Domains[1].Zones))
```

Comment:
is there a need for the commented printfs?... should it just be showed if extra verbose is required? or should it be completely deleted... Comments apply for the whole file

Reply:
I believe they were added temporarily for debugging the test case scenario but should not always be logged. If we decide to update the UT scenario and debug the UT they could be useful again. Since this is UT code I am okay with leaving them there for easy access in the future.